lib/SetImageChannelMetadataPass.cpp

```diff
@@ -0,0 +1,136 @@
+// Copyright 2022 The Clspv Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "llvm/IR/Instructions.h"
+#include "llvm/IR/Module.h"
+
+#include <map>
+
+#include "Builtins.h"
+#include "Constants.h"
+#include "PushConstant.h"
+#include "SetImageChannelMetadataPass.h"
+
+using namespace llvm;
+
+#define DEBUG_TYPE "setimagechannelmetadata"
+
+namespace {
+
+using ImageGetterMap =
+    std::map<std::pair<unsigned, unsigned>, SmallVector<CallInst *, 1>>;
+using MetadataVector = SmallVector<Metadata *, 3>;
+
+unsigned getImageOrdinal(Value *Image) {
+  auto *img_call = dyn_cast<CallInst>(Image);
+  assert(img_call != nullptr);
+  assert(clspv::Builtins::Lookup(img_call->getCalledFunction()).getType() ==
+         clspv::Builtins::kClspvResource);
+  return dyn_cast<ConstantInt>(
+             img_call->getOperand(clspv::ClspvOperand::kResourceArgIndex))
+      ->getZExtValue();
+}
+
+void concatWithFunctionMetadata(Function *F, MetadataVector &MDs) {
+  auto fct_md = F->getMetadata(clspv::PushConstantsMetadataImageChannelName());
+  if (fct_md != nullptr) {
+    for (unsigned i = 0; i < fct_md->getNumOperands(); i++) {
+      MDs.push_back(fct_md->getOperand(i));
+    }
+  }
+}
+
+unsigned setMetadata(Module &M, Function *F, ImageGetterMap &map) {
+  auto i32 = IntegerType::get(M.getContext(), 32);
+
+  unsigned int count = 0;
+  for (auto elem : map) {
```

Comment:
This is triggering: https://clang.llvm.org/extra/clang-tidy/checks/performance/for-range-copy.html 
'elem' can be const reference.